pkg/api/http_client.go

```diff
@@ -146,6 +153,14 @@ func normalizeHostname(hostname string) string {
 	if strings.HasSuffix(hostname, "."+localhost) {
 		return localhost
 	}
+	// This has been copied over from the cli/cli NormalizeHostname function
+	// to ensure compatible behaviour but we don't fully understand when or
+	// why it would be useful here. We can't see what harm will come of
+	// duplicating the logic.
+	if before, found := strings.CutSuffix(hostname, "."+tenancyHost); found {
+		idx := strings.LastIndex(before, ".")
+		return fmt.Sprintf("%s.%s", before[idx+1:], tenancyHost)
+	}
```

Comment:
Similar to my comment above, it seems we now have 3 copies of this logic across `cli/cli` and 2 places within `cli/go-gh` and that makes me a little concerned:

https://github.com/cli/go-gh/blob/25db6b99518c88e03f71dbe9e58397c4cfb62caf/pkg/auth/auth.go#L169-L186

Reply:
Same as my response above.